src/mod/l4_thruster.c

```diff
@@ -541,27 +741,1590 @@ void GeneralReadRequest(int argc, char *argv[]){
+	 *
+	 *
+	 */
+
+
+	LAST_STARTED_MODULE = 1108;
+	uint16_t procedure_id = atoi(argv[0]); // procedure_id is always fist index of argument array
+	uint32_t duration = atoi(argv[1]);
+	uint32_t logging_dt = atoi(argv[2]); //Time after which READ request is sent
+	uint32_t now_timestamp = (uint32_t)timGetSystime();
+	uint8_t register_index = atoi(argv[3]); // Register that is intended to be monitored
+	char print_str[200];
+	int len;
+	char *temp_argv[2];
+	int waiting_between_logging;
+	char argument1[50];
+
+	uint32_t wait_for_read_request_proccess = 3000; // WARNING ! Wait time for proccesing request [ms] should not be equal to logging dt [ms]
+	// Otherwise function newer exits
+
+
+	switch (THR_HARDCODED_SEQUENCES[procedure_id].substage_index){
+
+	case 0:
+
+		if ( ((now_timestamp - THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage) % logging_dt ) == 0 ){
+				waiting_between_logging = (int)( (now_timestamp - THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage)/1000 );
+
+				sprintf(print_str, "\n Waiting %d s \n",waiting_between_logging  );
+				len = strlen(print_str);
+				deb_print_pure_debug((uint8_t *)print_str, len);
+
+
+
+
+				//temp_argv[0]= "6";
+				//temp_argv[1]= "99"; // Reservoir Temperature
+				//GeneralReadRequest(2, temp_argv);
+
+				THR_HARDCODED_SEQUENCES[procedure_id].substage_index++; // increase substage to procede with READ request
+				THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_substage = (uint32_t)timGetSystime(); // save timestamp of substage completion
+				break;
+
+					}
+
+
+
+		if ( (now_timestamp - THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage) < duration ){
+				// do nothing
+			}
+		else{
+				THR_HARDCODED_SEQUENCES[procedure_id].sequence_execution_stage = (uint32_t)timGetSystime(); //save execution finish time of wait stage
+
+				sprintf(print_str, "\nWait and Monitor Stage = %d Complete t = %d\n", THR_HARDCODED_SEQUENCES[procedure_id].execution_index,now_timestamp);
+				len = strlen(print_str);
+				deb_print_pure_debug((uint8_t *)print_str, len);
+				THR_HARDCODED_SEQUENCES[procedure_id].execution_index++; // increase sequence execution index so that after wait - next module to be executed
+				THR_HARDCODED_SEQUENCES[procedure_id].substage_index = 0;
+
+				//*******assume that argv[5] is custom print message !!!!! WARNING I AM NOT SURE THAT THIS IS GOOD IDEA
+				len = strlen(argv[5]);
+				deb_print_pure_debug((uint8_t *)argv[5], len);
+				//// WARNING THIS BLOCK MIGHT BE NO GOOD *********
+			}
+
+
+
+
+		break;
+	case 1:
+		sprintf(print_str, "\n Sending READ [%d] Request %d\n",register_index, now_timestamp );
+		len = strlen(print_str);
+		deb_print_pure_debug((uint8_t *)print_str, len);
+
+		temp_argv[0]= "6";
+		//temp_argv[1]= "97"; // Reservoir Temperature
+
+		sprintf(argument1, "%d",register_index);
+		temp_argv[1]= argument1;
+
+		GeneralReadRequest(2, temp_argv);
```

Comment:
I would have to redesign all GeneralRequest functions to avoid this mess with sprintf() and argv ...

This is total stupid.  this argv approach was inherited from commands .... but totally redundent here. 

When I realized that this is stupid - it was too late and whole sequencer module was almost complete .... This is why I decided to "follow the style" to produce solution that works as fast as possible without argv redesign